M2/Macaulay2/packages/GKMManifolds/Documentation_GKMManifolds.m2

```diff
@@ -0,0 +1,2165 @@
+	Inputs
+		X:TVariety
+			the source T-variety of the map
+		Y:TVariety
+			the target T-variety of the map
+		L:List
+			of pairs (x,y) where x and y are members of @TT "X.points"@ and @TT "Y.points"@, respectively
+	Outputs
+		f:TMap
+	Description
+		Text
+			This method creates a @TO TMap@ given a T-Variety $X$, a T-variety $Y$,
+			and a list @TT "L"@ of pairs (x,y) where x and y are members of @TT "X.points"@ and @TT "Y.points"@ (respectively),
+			indicating that the T-fixed point x of X is sent to the T-fixed point y of Y under the map.
+
+		Text
+			The following describes the projection from the third Hizerbruch surface to the projective 
+			line.
+			
+		Example
+			R = makeCharRing 2;
+			F3 = tVariety(hirzebruchSurface 3,R);
+			PP1 = tProjectiveSpace(1,R);
+			L = {({0,1},set {0}), ({0,3}, set{0}), ({1,2}, set{1}), ({2,3}, set{1})};
+			f = tMap(F3,PP1,L)
+	Caveat
+		This does not check that the morphism is well defined. In particular, it does not
+		verify that the map on T-fixed points is induced by a morphism of T-varieties.						
+	SeeAlso
+		diagonalTMap
+		tFlagMap
+		(pullback, TMap)
+		pushforward
+		tChi
+///
+
+doc ///
+	Key
+		tFlagMap
+		(tFlagMap, TVariety, TVariety)
+	Headline
+		creates TMaps bewteen generalized flag varieties
+	Usage
+		f = tFlagMap(X,Y)
+	Inputs
+		X:TVariety
+			the source generalized flag variety
+		Y:TVariety
+			the target generalized flag variety
+	Outputs
+		f:TMap
+	Description
+		Text
+			Let $L =\{k_1,\dots,k_m\}$ be a set of ranks of linear subscpaces of $\mathbb C^n$ and consider
+			a subset $L' \subseteq L$. Let $X = Fl(L; n)$ and $Y=Fl(L';n)$ be the associated generalized 
+			flag varieties (if they exist). This method produces the canonical projection from $X$ to $Y$
+			that forgets the linear subspaces having ranks $L \setminus L'$.
+		Example
+			R = makeCharRing 3
+		   	X = tGeneralizedFlagVariety("B",3,{1,2},R);
+			Y1 = tGeneralizedFlagVariety("B",3,{2},R);
+			Y2 = tGeneralizedFlagVariety("B",3,{1},R);
+			peek tFlagMap(X,Y1)
+			peek tFlagMap(X,Y2)
+	SeeAlso
+		tMap
+		diagonalTMap
+		tGeneralizedFlagVariety
+
+///
+
+
+
+
+doc ///
+	Key
+		(pullback, TMap)
+	Headline
+		computes the pullback map of T-equivariant K-classes of a T-map
+	Usage
+		pullback(f)
+	Inputs
+		f:TMap
+	Outputs
+		:FunctionClosure
+			whose input is a @TO TKClass@ on the target @TO TVariety@ of f and output is its pullback along f
+	Description
+		Text
+			Given two $T$-varieties $X$ and $Y$, this method computes the pullback of a @TO TKClass@ on $Y$ 
+			along a T-equivariant morphism $X \to Y$.
+
+		Example
+			R = makeCharRing 4;
+			FlGr = tGeneralizedFlagVariety("A",3,{1,2},R)
+			Gr24 = tGeneralizedFlagVariety("A",3,{2},R)
+			f = tFlagMap(FlGr,Gr24)
+			O1 = ampleTKClass Gr24
+			(pullback f)(O1)
+			
+	SeeAlso
+		tFlagMap
+		pushforward
+///
+
+
+doc ///
+	Key
+		pushforward
+		(pushforward, TMap)
+	Headline
+		computes the pushforward map of T-equivariant K-classes of a T-map
```

Comment:
starting so many headlines with "computes the" seems redundant, since every function computes something.